cpp/src/arrow/dataset/filter.cc

```diff
@@ -704,28 +704,38 @@ using arrow::internal::JoinStrings;
 
 std::string AndExpression::ToString() const {
   return JoinStrings(
-      {"(", left_operand_->ToString(), " and ", right_operand_->ToString(), ")"}, "");
+      std::vector<util::string_view>{"(", left_operand_->ToString(), " and ",
+                                     right_operand_->ToString(), ")"},
+      "");
```

Comment:
@bkietz Do tell if this is too verbose for your taste (to be honest, I'm not sure why `JoinStrings` is used for simple concatenation here).